src/ImuSensor.cc

```diff
@@ -292,6 +285,102 @@ math::Pose3d ImuSensor::WorldPose() const
   return this->dataPtr->worldPose;
 }
 
+//////////////////////////////////////////////////
+void ImuSensor::ReadLocalizationTag(const sdf::Sensor &_sdf)
+{
+  std::string localization = _sdf.ImuSensor()->Localization();
+
+  if (localization == "ENU")
+  {
+    this->dataPtr->sensorOrientationRelativeTo = WorldFrameEnumType::ENU;
+  } else if (localization == "NED") {
+    this->dataPtr->sensorOrientationRelativeTo = WorldFrameEnumType::NED;
+  } else if (localization == "NWU") {
+    this->dataPtr->sensorOrientationRelativeTo = WorldFrameEnumType::NWU;
+  } else if (localization == "CUSTOM") {
+    this->dataPtr->sensorOrientationRelativeTo = WorldFrameEnumType::CUSTOM;
+  } else {
+    this->dataPtr->sensorOrientationRelativeTo = WorldFrameEnumType::NONE;
+  }
+
+  // Set orientation reference frame if custom_rpy was supplied
+  if (this->dataPtr->sensorOrientationRelativeTo == WorldFrameEnumType::CUSTOM)
+  {
+    if (_sdf.ImuSensor()->CustomRpyParentFrame() == "")
+    {
+      this->SetOrientationReference(ignition::math::Quaterniond(
+        _sdf.ImuSensor()->CustomRpy()));
+    }
+    else
+    {
+      ignwarn << "custom_rpy parent frame must be set to empty "
+                "string. Setting it to any other frame is not "
+                "supported yet." << std::endl;
+    }
+  }
+}
+
+
+//////////////////////////////////////////////////
+void ImuSensor::ReadLocalizationTag(sdf::ElementPtr _sdf)
+{
+  sdf::Sensor sdfSensor;
+  sdfSensor.Load(_sdf);
+  return this->ReadLocalizationTag(sdfSensor);
+}
+
+//////////////////////////////////////////////////
+void ImuSensor::SetWorldFrameOrientation(
+  const math::Quaterniond &_rot, WorldFrameEnumType _relativeTo)
+{
+  this->dataPtr->worldRelativeOrientation = _rot;
+  this->dataPtr->worldFrameRelativeTo = _relativeTo;
+
+  // Table to hold frame transformations
+  std::map<WorldFrameEnumType,
```

Comment:
this table consists of constant values. I think it could be `static` and `const`?

Reply:
Done